storage/badger/operation/common.go

```diff
@@ -564,3 +565,30 @@ func Fail(err error) func(*badger.Txn) error {
 		return err
 	}
 }
+
+func encodeAndCompress(entity interface{}) ([]byte, error) {
+	// serialize the entity data
+	val, err := msgpack.Marshal(entity)
+	if err != nil {
+		return nil, irrecoverable.NewExceptionf("could not encode entity: %w", err)
+	}
+
+	// compress the serialized data using Snappy
+	compressedVal := snappy.Encode(nil, val)
```

Comment:
Ideally there would be a flag controlling this compression, otherwise merging this into master is a pretty big breaking change right?

Reply:
Yes, will use flag to control